File: app/src/androidTest/java/de/andicodes/vergissnix/data/DatabaseTest.java
package de.andicodes.vergissnix.data;

import android.content.Context;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;

import java.time.ZoneOffset;
import java.time.ZonedDateTime;
import java.util.List;

import androidx.room.Room;
import androidx.test.core.app.ApplicationProvider;
import androidx.test.ext.junit.runners.AndroidJUnit4;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(AndroidJUnit4.class)
public class DatabaseTest {
    private TaskDao taskDao;
    private AppDatabase db;

    @Before
    public void createDb() {
        Context context = ApplicationProvider.getApplicationContext();
        db = Room.inMemoryDatabaseBuilder(context, AppDatabase.class).build();
        taskDao = db.taskDao();
    }

    @After
    public void closeDb() {
        db.close();
    }

    @Test
    public void writeTaskAndReadInList() throws Exception {
        ZonedDateTime time = ZonedDateTime.of(2021, 11, 6, 10, 29, 57, 632, ZoneOffset.ofHours(1));
        Task task = new Task();
        task.setText("Test Task");
        task.setTime(time);

        Task savedTask = taskDao.saveTask(task);

        List<Task> tasks = taskDao.allTasks().getValue();
        assertThat(tasks).containsExactly(savedTask);
    }
}

File: app/src/androidTest/java/de/andicodes/vergissnix/data/DatabaseTest.kt
package de.andicodes.vergissnix.data

import android.content.Context
import androidx.room.Room
import androidx.test.core.app.ApplicationProvider
import androidx.test.ext.junit.runners.AndroidJUnit4
import kotlinx.coroutines.flow.first
import kotlinx.coroutines.test.runTest
import org.assertj.core.api.Assertions
import org.junit.After
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import java.time.ZonedDateTime
import java.time.temporal.ChronoUnit

@RunWith(AndroidJUnit4::class)
class DatabaseTest {
    private var taskDao: TaskDao? = null
    private var db: AppDatabase? = null

    @Before
    fun createDb() {
        val context = ApplicationProvider.getApplicationContext<Context>()
        db = Room.inMemoryDatabaseBuilder(context, AppDatabase::class.java).build()
        taskDao = db!!.taskDao()
    }

    @After
    fun closeDb() {
        db!!.close()
    }

    @Test
    fun writeTaskAndReadInList() = runTest {
        val task = Task()
        task.text = "Test Task"
        task.time = ZonedDateTime.parse("2021-11-06T10:29:57+01:00")

        val savedTask = taskDao!!.saveTask(task)
        val tasks = taskDao!!.allTasks().first()
        Assertions.assertThat(tasks)
            .isNotNull()
            .hasSize(1)
        Assertions.assertThat(tasks[0]).usingRecursiveComparison()
            .ignoringFields("timeCreated")
            .isEqualTo(savedTask)
        Assertions.assertThat(tasks[0].timeCreated)
            .isCloseTo(savedTask.timeCreated, Assertions.within(0, ChronoUnit.SECONDS))
    }
}